bokehjs/gulp/linker.ts

```diff
@@ -342,7 +351,52 @@ export class Module {
     })
   }
 
-  get source(): string {
+  remove_esmodule(): void {
+    const self = this
+    const body: typeof self.ast.body = []
+    for (const node of this.ast.body) {
+      if (node.type == "ExpressionStatement") {
+        const expr = node.expression
+        if (expr.type == "Literal") {
+          if (expr.value == "use strict")
+            continue
+        }
+        if (expr.type == "CallExpression") {
+          const args = expr.arguments
+          if (args.length == 3) {
+            const arg = args[1]
+            if (arg.type == "Literal") {
+              if (arg.value == "__esModule")
+                continue
+            }
+          }
+        }
+      }
+      body.push(node)
+    }
+    this.ast.body = body
+  }
+
+  wrap_in_function(): void {
+    const wrapper: Statement = {
+      type: "FunctionDeclaration",
+      id: {type: "Identifier", name: "_"},
+      params: [
+        {type: "Identifier", name: "require"},
+        {type: "Identifier", name: "module"},
+        {type: "Identifier", name: "exports"},
+      ],
+      body: {
+        type: "BlockStatement",
+        body: Array.from(this.ast.body) as Statement[],
+      }
+    }
+    const comment: Comment = {type: "Block", value: this.canonical}
+    wrapper.leadingComments = [comment]
+    this.ast.body = [wrapper]
+  }
```

Comment:
Presumably this can go away when the port is fully completed?

Reply:
No, this has nothing to do with coffeescript or typescript. Given that module ids aren't human readable (or readable at all in the case of the main bundle), this is a quick way to ease reading bundle contents.

Reply:
You mentioned having to work around something about how CS2 converted  classes to closures (?) where is that? (just curious)

Reply:
That's a different piece of code:
https://github.com/bokeh/bokeh/blob/b8b7559cd67da9a43ec950b2fa090959ba9e619b/bokehjs/gulp/tasks/scripts.ts#L31-L57

Reply:
Yeah, the above garbage will go away as soon as transition is done.